pkg/backends/influxdb/influxdb.go

```diff
@@ -0,0 +1,502 @@
+	fl := flush{
+		timestampSeconds: nowSeconds,
+		flushIntervalSec: idb.flushInterval.Seconds(),
+		metricsPerBatch:  idb.metricsPerBatch,
+		disabledSubtypes: idb.disabledSubtypes,
+		errorCounter:     &idb.batchesCreateFailed,
+		cb:               cb,
+		getBuffer: func() (*bytes.Buffer, io.WriteCloser) {
+			return idb.getBuffer(ctx)
+		},
+		releaseBuffer: idb.releaseBuffer,
+	}
+
+	fl.buffer, fl.writer = fl.getBuffer()
+
+	metrics.Counters.Each(func(metricName, tagsKey string, counter gostatsd.Counter) {
+		if fl.buffer == nil {
+			return
+		}
+		fl.addCounter(metricName, counter.Tags, counter.Value, counter.PerSecond)
+	})
+
+	metrics.Timers.Each(func(metricName, tagsKey string, timer gostatsd.Timer) {
+		if fl.buffer == nil {
+			return
+		}
+		if timer.Histogram == nil {
+			fl.addBaseTimer(metricName, timer)
+		} else {
+			fl.addHistogramTimer(metricName, timer)
+		}
+	})
+
+	metrics.Gauges.Each(func(metricName, tagsKey string, g gostatsd.Gauge) {
+		if fl.buffer == nil {
+			return
+		}
+		fl.addGauge(metricName, g.Tags, g.Value)
+	})
+
+	metrics.Sets.Each(func(metricName, tagsKey string, set gostatsd.Set) {
+		if fl.buffer == nil {
+			return
+		}
+		fl.addSet(metricName, set.Tags, uint64(len(set.Values)))
+	})
+
+	if fl.metricCount == 0 {
+		idb.releaseBuffer(fl.buffer)
+		return
+	}
+
+	fl.finish()
+}
+
+func (idb *Client) postData(ctx context.Context, buffer *bytes.Buffer, seriesCount uint64) error {
+	if err := idb.post(ctx, buffer); err != nil {
+		return err
+	}
+	atomic.AddUint64(&idb.seriesSent, seriesCount)
+	return nil
+}
+
+func writeEvent(w io.Writer, e *gostatsd.Event) {
+	hasHost := false
+	var eventTags []string
+	tags := make(gostatsd.Tags, 0, len(e.Tags)+4)
+	for _, tag := range e.Tags {
+		if strings.HasPrefix(tag, "host:") {
+			hasHost = true
+		}
+		if !strings.HasPrefix(tag, "eventtags") {
+			tags = append(tags, tag)
+		} else {
+			parts := strings.SplitN(tag, ":", 2)
+			if len(parts) > 1 {
+				eventTags = append(eventTags, parts[1])
+			}
+		}
+	}
+
+	// We don't need to use .Concat(), because we created the original slice.
+	tags = append(
+		tags,
+		"priority:"+e.Priority.String(),
+		"alerttype:"+e.AlertType.String(),
+	)
+	if e.SourceTypeName != "" {
+		tags = append(tags, "sourcetypename:"+e.SourceTypeName)
+	}
+	if e.Hostname != "" && !hasHost {
+		tags = append(tags, "host:"+e.Hostname)
+	}
+
+	writeName(w, "events", tags)
+
+	sb := &strings.Builder{}
+
+	sb.WriteString("title=")
+	escapeStringToBuilder(sb, e.Title)
+	sb.WriteString(",text=")
+	escapeStringToBuilder(sb, e.Text)
+	if len(eventTags) > 0 {
+		sb.WriteString(",tags=")
+		escapeStringToBuilder(sb, strings.Join(eventTags, ","))
+	}
+	sb.WriteString(fmt.Sprintf(" %d\n", e.DateHappened))
+	_, _ = w.Write([]byte(sb.String()))
+}
+
+func (idb *Client) SendEvent(ctx context.Context, e *gostatsd.Event) error {
+	buf, writer := idb.getBuffer(ctx)
+	if buf == nil {
+		return ctx.Err()
+	}
+	defer idb.releaseBuffer(buf)
+
+	writeEvent(writer, e)
```

Comment:
The write error should be captured here

Reply:
I have a method for handling errors, which I'm tentatively calling "draeaswh", or "don't return an error, and see what happens".

That's why `writeEvent` has no error return value, but we do grab it from `writer.Close()` on the next line.